app/services/course/assessment/question/programming/cpp/cpp_package_service.rb

```diff
@@ -0,0 +1,254 @@
+# frozen_string_literal: true
+class Course::Assessment::Question::Programming::Cpp::CppPackageService < \
+  Course::Assessment::Question::Programming::LanguagePackageService
+  AUTOGRADE_INCLUDE_PATH = File.join(File.expand_path(File.dirname(__FILE__)),
+                                 'cpp_autograde_include.cc').freeze
+
+  AUTOGRADE_PRE_PATH = File.join(File.expand_path(File.dirname(__FILE__)),
+                                 'cpp_autograde_pre.cc').freeze
+
+  AUTOGRADE_POST_PATH = File.join(File.expand_path(File.dirname(__FILE__)),
+                                  'cpp_autograde_post.cc').freeze
+
+  MAKEFILE_PATH = File.join(File.expand_path(File.dirname(__FILE__)), 'cpp_makefile').freeze
+
+  def autograded?
```

Comment:
I think we can move this to service superclass also. Make it a convention for services of all languages.